src/loki/search.cc

```diff
@@ -386,8 +386,13 @@ struct bin_handler_t {
       if (!candidate.edge->forward()) {
         length_ratio = 1.f - length_ratio;
       }
+      // side of street TODO: set limitations on when we should use provided display ll to control
       // side of street
-      auto side = candidate.get_side(location.latlng_, candidate.sq_distance);
+      auto side =
+          candidate.get_side(location.display_latlng_ ? *location.display_latlng_ : location.latlng_,
+                             location.display_latlng_
+                                 ? location.display_latlng_->DistanceSquared(candidate.point)
+                                 : candidate.sq_distance);
```

Comment:
this is the meat of the change. if we provided a display ll we use that in conjunction with its distance from the candidate/snapped location to determine side of street, otherwise we fall back to the previous logic which was to use the input location ll as the display ll as well as the routing ll and its distance from the candidate/snapped location

Reply:
we really should handle that `TODO` before we ship this need to worry about the case where the display ll isnt really near the route ll